apis/v1alpha1/shared_types.go

```diff
@@ -194,6 +194,37 @@ type RouteGatewayStatus struct {
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 }
 
+// GatewayStatusReference identifies a Gateway in a specified namespace. This
+// reference also includes a controller name to simplify cleaning up status
+// entries.
+type GatewayStatusReference struct {
```

Comment:
From Go API user's perspective, I can confuse this as a reference of something inside GatewayStatus.
How about `RouteStatusGatewayReference`?

This is a little bit of a style and language thing so hard to concrete argue one way or the other.

Reply:
Yeah good call, that is better. Updated to use that name.